app/features-json/build_json_controller.rb

```diff
@@ -92,6 +84,47 @@ class BuildJSONController < APIController
       json(build_summary_view_model)
     end
 
+    get "#{HOME}/:build_number/logs" do |project_id, build_number|
+      # `current_build_runner` is only defined if the build was just run a while back
+      # if the server was restarted, we're gonna end here in this code block
+      build_log_artifact = current_build.artifacts.find do |current_artifact|
+        # We can improve the detection in the future, to actually mark an artifact as "default output"
+        current_artifact.type.include?("log") && current_artifact.reference.end_with?("fastlane.log")
+      end
+
+      if build_log_artifact
+        artifact_file_content = File.read(build_log_artifact.provider.retrieve!(artifact: build_log_artifact))
```

Comment:
This will work for local file storage only. As with `gcp` artifact storage provider we'll return a URL that we'll need to fetch. This doesn't have to be addressed in this PR, however let's either add a `# TODO` here or create a GH issue to follow up 👍 I'm sure we have other work to do for the GCP integration

Reply:
added TODO